client/src/components/FormBuilder/FormBuilder.js

```diff
@@ -113,9 +113,16 @@ class FormBuilder extends Component {
     delete componentProps.input;
     const { identifier } = this.props;
     const { name } = componentProps;
+    const context = [
+      identifier,
+      componentProps.data.extraContext,
```

Comment:
this seems like a bad practice and also not mentioned in the pull request, why isn't the context in the identifier where it is easily discoverable?

So you're looking to infer an extra dynamic context right?
I remember we removed dynamic contexts due to performance hits it potentially brings to the table (especially because props can and will change dynamically with transformers).

Reply:
Hmm, you're completely right. Let me edit the PR context instead of elaborating here.

Reply:
hmm, I don't think `extraContext` is the proper name for this property described then.

I can understand the need for something like the usecase but why can't the transformer check for other data that you've provided such as `ElementIcon`, `ElementID` and `ElementID`? And if those data points don't match then pass-through the original FieldGroup.

I see that you've already got a filter for the transformer, so you're not too far off from that, I think?

Reply:
don't get me wrong, I'm looking for a justification for this potential performance hit :)

Reply:
bump @NightJar

Reply:
Naming things is hard. The intention here was such that the `contextGroup` (aka extraContext) does not need to be dynamic. The data array was readily available but may not be the correct place for it I can agree. However I think having some form of grouping elements via context should something provided via the core. Ideally I guess grouping would be a part of the FormField's `getSchemaState` so that it comes out in `this.props` as opposed to `componentProps`, but that would require a third dependency PR to framework. Is this something you'd see as a way forward, or do you think the thing you mentioned with the transformer filter that I didn't really understand (please elaborate) is better?

Reply:
Yeah cool, 'short circuiting' the render function is what I thought you meant, thanks for clarifying/confirming :)

I'm open to the idea of structuring the other PR another way, the subclassing was more the immediate way I could see that would allow me to alter the output of the render function. I had originally tried specifying a different component to use (via schema definition), however that didn't seem to work, and using a transform seemed better in terms of approach anyway.

I'll fiddle around a bit and see if I can maintain functionality whilst reducing this PR to only cover the fieldHolder alterations (i.e. drop the context alterations).

Reply:
Cool, I just got rid of all this stuff :)
Thanks for helping me through it @flamerohr